src/js/components/sidebar.jsx

```diff
@@ -402,6 +455,14 @@ export default class SideBar extends React.Component {
       }, []);
   }
 
+  _hasPinnedTabs(filteredTabs) {
+    const { tabsById } = this.state;
+
+    return filteredTabs.find(id => tabsById.get(id).pinned) == undefined
```

Comment:
Like above, I think you can use `filteredTabs.some` here and get the same effect with a simpler syntax.